### ISSUE TITLE
Game won't load KShoot INPUT GATE charts

### ISSUE THREAD

**Issue by Dan9er** [Drewol/unnamed-sdvx-clone#7]
I am using the version via the download link.

### File Tree
```
songs
├───hybridintermission
│       gate.lst
│       hybridintermission.jpg
│       hybridintermission.ogg
│       hybridintermission.wmv
│       hybridintermission_ch.ksh
│       hybridintermission_ex.ksh
│       hybridintermission_lt.ksh
│
└───theordeal
        gate.lst
        theordeal.jpg
        theordeal.ogg
        theordeal_ch.ksh
        theordeal_ex.ksh
        theordeal_in.jpg
        theordeal_in.ksh
        theordeal_lt.ksh
```

### Log from trying to play *Hybrid Intermission*, then *The Ordeal*
BTW, the log was originally ALL on one line. So is the Readme files. Please fix that too.
```
[17:02:27][Info] Starting task "Application Setup"
[17:02:28][Warning] No joysticks found
[17:02:28][Info] Starting task "Audio Init"
[17:02:28][Info] Finished task "Audio Init" in  65 ms
[17:02:28][Info] Starting task "GL Init"
[17:02:28][Info] OpenGL Version: 4.5.13491 Compatibility Profile Context 22.19.662.4
[17:02:28][Info] Finished task "GL Init" in  118 ms
[17:02:28][Info] Listing shader uniforms for shaders\font.vs
[17:02:28][Info] Uniform [0, loc=0, Unknown [35676]] = proj
[17:02:28][Info] Uniform [1, loc=1, Unknown [35676]] = world
[17:02:28][Info] Listing shader uniforms for shaders\font.fs
[17:02:28][Info] Uniform [0, loc=0, Unknown [35666]] = color
[17:02:28][Info] Uniform [1, loc=1, Unknown [35678]] = mainTex
[17:02:28][Info] Listing shader uniforms for shaders\guiTex.vs
[17:02:28][Info] Uniform [0, loc=0, Unknown [35676]] = proj
[17:02:28][Info] Uniform [1, loc=1, Unknown [35676]] = world
[17:02:28][Info] Listing shader uniforms for shaders\guiTex.fs
[17:02:28][Info] Uniform [0, loc=0, Unknown [35666]] = color
[17:02:28][Info] Uniform [1, loc=1, Unknown [35678]] = mainTex
[17:02:28][Info] Listing shader uniforms for shaders\guiColor.vs
[17:02:28][Info] Uniform [0, loc=0, Unknown [35676]] = proj
[17:02:28][Info] Uniform [1, loc=1, Unknown [35676]] = world
[17:02:28][Info] Listing shader uniforms for shaders\guiColor.fs
[17:02:28][Info] Uniform [0, loc=0, Unknown [35666]] = color
[17:02:28][Info] Listing shader uniforms for shaders\guiButton.vs
[17:02:28][Info] Listing shader uniforms for shaders\guiButton.fs
[17:02:28][Info] Uniform [0, loc=0, Unknown [35666]] = color
[17:02:28][Info] Uniform [1, loc=1, Unknown [35678]] = mainTex
[17:02:28][Info] Listing shader uniforms for shaders\guiButton.gs
[17:02:28][Info] Uniform [0, loc=0, Unknown [35666]] = border
[17:02:28][Info] Uniform [1, loc=1, Unknown [35676]] = proj
[17:02:28][Info] Uniform [2, loc=2, Unknown [35664]] = size
[17:02:28][Info] Uniform [3, loc=3, Unknown [35666]] = texBorder
[17:02:28][Info] Uniform [4, loc=4, Unknown [35676]] = world
[17:02:28][Info] Listing shader uniforms for shaders\guiGraph.vs
[17:02:28][Info] Uniform [0, loc=0, Unknown [35676]] = proj
[17:02:28][Info] Uniform [1, loc=1, Unknown [35676]] = world
[17:02:28][Info] Listing shader uniforms for shaders\guiGraph.fs
[17:02:28][Info] Uniform [0, loc=0, Unknown [35667]] = viewport
[17:02:28][Info] Uniform [1, loc=1, Unknown [35678]] = graphTex
[17:02:28][Info] Finished task "Application Setup" in  872 ms
[17:02:30][Info] Listing shader uniforms for shaders\diffFrame.vs
[17:02:30][Info] Uniform [0, loc=0, Unknown [35676]] = proj
[17:02:30][Info] Uniform [1, loc=1, Unknown [35676]] = world
[17:02:30][Info] Listing shader uniforms for shaders\diffFrame.fs
[17:02:30][Info] Uniform [0, loc=0, Unknown [5126]] = selected
[17:02:30][Info] Uniform [1, loc=1, Unknown [5126]] = time
[17:02:30][Info] Uniform [2, loc=2, Unknown [35678]] = frame
[17:02:30][Info] Uniform [3, loc=3, Unknown [35678]] = jacket
[17:02:30][Info] Starting task "Map Database - Enumerate Files and Folders"
[17:02:30][Info] Finished task "Map Database - Enumerate Files and Folders" in  1 ms
[17:02:30][Info] Starting task "Map Database - Process Removed Files"
[17:02:30][Info] Finished task "Map Database - Process Removed Files" in  0 ms
[17:02:30][Info] Starting task "Map Database - Process New Files"
[17:02:30][Info] Finished task "Map Database - Process New Files" in  0 ms
[17:02:30][Info] Timing restart, delta = -65.490149
[17:02:30][Info] Timing restart, delta = -60.704600
[17:02:36][Info] Starting task "AsyncLoad Game"
[17:02:36][Info] Starting task "Load Beatmap"
[17:02:36][Info] Starting task "Load KShootMap"
[17:02:36][Info] Finished task "Load KShootMap" in  66 ms
[17:02:36][Warning] [KSH]Unkown map parameter at 43:0: stop
[17:02:36][Info] Finished task "Load Beatmap" in  68 ms
[17:02:36][Info] Reseting BeatmapPlayback with StartTime = 0
[17:02:36][Error] FX audio for for beatmap does not exists at: "songs\hybridintermission\hybridintermission_lt_f.ogg"
[17:02:36][Info] Finished task "AsyncLoad Game" in  81 ms
[17:02:36][Error] [Transition] Failed to load tickable
[17:02:36][Error] [Transition] Failed to load tickable
[17:02:36][Info] Starting task "Map Database - Enumerate Files and Folders"
[17:02:36][Info] Finished task "Map Database - Enumerate Files and Folders" in  1 ms
[17:02:36][Info] Starting task "Map Database - Process Removed Files"
[17:02:36][Info] Finished task "Map Database - Process Removed Files" in  0 ms
[17:02:36][Info] Starting task "Map Database - Process New Files"
[17:02:36][Info] Finished task "Map Database - Process New Files" in  0 ms
[17:02:36][Info] Timing restart, delta = -60.706864
[17:02:38][Info] Timing restart, delta = -65.494540
[17:02:43][Info] Starting task "AsyncLoad Game"
[17:02:43][Info] Starting task "Load Beatmap"
[17:02:43][Info] Starting task "Load KShootMap"
[17:02:43][Info] Finished task "Load KShootMap" in  36 ms
[17:02:43][Info] Finished task "Load Beatmap" in  38 ms
[17:02:43][Info] Reseting BeatmapPlayback with StartTime = 0
[17:02:43][Error] FX audio for for beatmap does not exists at: "songs\theordeal\theordeal_lt_f.ogg"
[17:02:43][Info] Finished task "AsyncLoad Game" in  48 ms
[17:02:43][Error] [Transition] Failed to load tickable
[17:02:43][Error] [Transition] Failed to load tickable
[17:02:43][Info] Starting task "Map Database - Enumerate Files and Folders"
[17:02:43][Info] Finished task "Map Database - Enumerate Files and Folders" in  1 ms
[17:02:43][Info] Starting task "Map Database - Process Removed Files"
[17:02:43][Info] Finished task "Map Database - Process Removed Files" in  0 ms
[17:02:43][Info] Starting task "Map Database - Process New Files"
[17:02:43][Info] Finished task "Map Database - Process New Files" in  0 ms
[17:02:43][Info] Timing restart, delta = -65.497563
[17:02:46][Info] Starting task "Application Cleanup"
[17:02:46][Info] Cleaned up 19 resource(s) of class Graphics::MeshRes
[17:02:46][Info] Cleaned up 27 resource(s) of class Graphics::TextureRes
[17:02:46][Info] Cleaned up 13 resource(s) of class Graphics::ShaderRes
[17:02:46][Info] Cleaned up 1 resource(s) of class Graphics::FontRes
[17:02:46][Info] Cleaned up 6 resource(s) of class Graphics::MaterialRes
[17:02:48][Info] Finished task "Application Cleanup" in  1846 ms
```

**Comment by Drewol** [comment]
This has to do with how the older official input gate songs were converted from pre-rendered effects to using real-time effects. They did the conversion by just removing the effected file but they didn't remove the part in the .ksh that says to use pre-rendered effects so unnamed-clone looks for those and can't find them and it sees that as an error. I will change it to just fall back to real-time effects if the specified effected file can not be found.

**Comment by Dan9er** [comment]
Alright. Just put `fixes #7` in the commit description when you fix it.

BTW, my Discord Tag is Dan9er#6633. I really want to help by making concept art. If you need something, ping me.
